runtime/jvmMain/src/kotlinx/serialization/JvmResolving.kt

```diff
@@ -6,30 +6,43 @@ package kotlinx.serialization
 
 import kotlinx.serialization.internal.*
 import java.lang.reflect.*
-import kotlin.reflect.KClass
+import kotlin.reflect.*
 
 @PublishedApi
 internal open class TypeBase<T>
 
+/**
+ * Provides an instance of [java.lang.reflect.Type] with
+ * generic arguments filled in.
+ *
+ * Consider using Kotlin's [typeOf] and [KType] since they can
+ * also provide information about nullability.
+ */
+@Deprecated("Consider using Kotlin type token instead", ReplaceWith("typeOf()"), level = DeprecationLevel.WARNING)
 inline fun <reified T> typeTokenOf(): Type {
     val base = object : TypeBase<T>() {}
     val superType = base::class.java.genericSuperclass!!
     return (superType as ParameterizedType).actualTypeArguments.first()!!
 }
 
 /**
- * This method uses reflection to construct serializer for given type. However,
- * since it accepts type token, it is available only on JVM by design,
+ * This method uses reflection to construct serializer for given type.
+ *
+ * However, since it accepts type token, it is available only on JVM by design,
  * and it can work correctly even with generics, so
  * it is not annotated with [ImplicitReflectionSerializer].
  *
+ * Consider using Kotlin's [typeOf], [KType] and [serializer] since they can
+ * also provide information about nullability.
+ *
  * Keep in mind that this is a 'heavy' call, so result probably should be cached somewhere else.
  *
  * This method intended for static, format-agnostic resolving (e.g. in adapter factories) so context is not used here.
  */
 @Suppress("UNCHECKED_CAST")
 @UseExperimental(ImplicitReflectionSerializer::class)
-fun serializerByTypeToken(type: Type): KSerializer<Any> = when (type) {
+// not deprecated because many converters (e.g. retrofit2-kotlinx-serialization-converter) still have to use java's Type
```

Comment:
This comment is a bit redundant, especially before the method body. The same thing can be rephrased in the documentation itself. 

Notes on KDoc:
* "However, since it accepts type token, it is available only on JVM by design" is also redundant, especially for functions from JVM module that have `java.lang.reflect.Type` in its parameters.
* "This method uses X to do Y" -> Does Y" or "Does Y using X"
* "This method intended for static, format-agnostic resolving (e.g. in adapter factories)"  lacking the context what are adapter factories

Fixed with my commit